tests/chainer_tests/functions_tests/math_tests/test_basic_math.py

```diff
@@ -245,41 +258,45 @@ def setUp(self):
         self.ggx2 = numpy.random.uniform(-1, 1, self.shape).astype(self.dtype)
         self.ggx3 = numpy.random.uniform(-1, 1, self.shape).astype(self.dtype)
 
-    def check_forward(self, func, x1_data, x2_data, x3_data):
+    def check_forward(self, func, x1_data, x2_data, x3_data, backend_config):
```

Comment:
You need to manually transfer all inputs to GPU if `backend_config.use_cuda` is `True`.
https://github.com/chainer/chainer/pull/4277/files?diff=split#diff-50ce841c2d595fcd3d6e74f9c363c260R60
Same for `check_backward` and `check_double_backward`.